File: Java/28_Sorting/LC_1333_FilterRestaurantsByVeganFriendlyPriceAndDistance.java
class Solution {
    public List<Integer> filterRestaurants(int[][] restaurants, int veganFriendly, int maxPrice, int maxDistance) {
        Arrays.sort(restaurants, (a, b) -> a[1] == b[1] ? b[0] - a[0] : b[1] - a[1]);
        List<Integer> res = new ArrayList<>();
        for (int[] r : restaurants) {
            if (r[2] >= veganFriendly && r[3] <= maxPrice && r[4] <= maxDistance) {
                res.add(r[0]);
            }
        }

        return res;
    }
}

public class LC_1333_FilterRestaurantsByVeganFriendlyPriceAndDistance {

}

File: Swift/Leetcode.playground/Sources/28_Sorting/LC_1333_FilterRestaurantsByVeganFriendlyPriceAndDistance.swift
class Solution {
    func filterRestaurants(_ restaurants: [[Int]], _ veganFriendly: Int, _ maxPrice: Int, _ maxDistance: Int) -> [Int] {
        var sortedRestaurants = restaurants.sorted(by: { $0[1] == $1[1] ? $1[0] < $0[0] : $1[1] < $0[1] })
        var res = [Int]()
        for r in sortedRestaurants {
            if r[2] >= veganFriendly && r[3] <= maxPrice && r[4] <= maxDistance {
                res.append(r[0])
            }
        }
        return res
    }
}